src/Tizen.NUI/src/internal/WebView/WebHttpRequestInterceptor.cs

```diff
@@ -46,56 +51,143 @@ public string Url
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public bool AddResponseHeader(string fieldName, string fieldValue)
         {
-            bool result = Interop.WebHttpRequestInterceptor.AddResponseHeader(SwigCPtr, fieldName, fieldValue);
+            bool result = Interop.WebHttpRequestInterceptor.AddResponseHeader(interceptorHandle, fieldName, fieldValue);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return result;
+        }
+
+        /// <summary>
+        /// Adds HTTP headers to response for intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
+        /// <param name="headers">Map of response headers</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool AddResponseHeaders(PropertyMap headers)
+        {
+            bool result = Interop.WebHttpRequestInterceptor.AddResponseHeaders(interceptorHandle, PropertyMap.getCPtr(headers));
             if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
             return result;
         }
 
         /// <summary>
         /// Writes whole response body at once.
+        /// To call it, application should have full response body ready for the intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
+        /// After this call, any further call on WebHttpRequestInterceptor results in undefined behavior.
+        /// <param name="body">Contents of response</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool SetResponseBody(string body)
+        {
+            if (body == null)
+                return false;
+            bool result = Interop.WebHttpRequestInterceptor.AddResponseBody(interceptorHandle, body, (uint)body.Length);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return result;
+        }
+
+        /// <summary>
+        /// Writes whole response body with headers at once.
+        /// To call it, application should have full response headers and body ready for the intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
+        /// After this call, any further call on WebHttpRequestInterceptor results in undefined behavior.
+        /// <param name="headers">Headers of response</param>
         /// <param name="body">Contents of response</param>
-        /// <param name="length">Length of Contents of response</param>
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public bool AddResponseBody(string body, uint length)
+        public bool SetResponse(string headers, string body)
```

Comment:
I wonder different of between headers on this method and `public bool AddResponseHeaders(PropertyMap headers)`

if they was semantically same, why not use same type?

Reply:
As I checked, this headers maybe including response status

Reply:
Yes. headers of http response need include a line like "HTTP/1.1 200 OK".